pkg/apis/events/validation/validation.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/core"
+	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
+)
+
+const (
+	reportingInstanceLengthLimit = 128
+	actionLengthLimit            = 128
+	reasonLengthLimit            = 128
+	noteLengthLimit              = 1024
+)
+
+// ValidateEvent validates new created events.
+func ValidateEvent(event *core.Event) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&event.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
```

Comment:
FTR - event names are somewhat special:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/event_recorder.go#L75

Though the validation looks fine.

Reply:
Should we add a specific check for the event name in ValidateEventSpec func?

Reply:
This is implementation detail, so let's leave as is.

Reply:
Yes - there isn't. But I think that's fine.

Reply:
Got it.